XMLtoMD/GuidelineXmlToMD/MarkdownOut/MdWriter.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.IO;
+
+namespace MarkdownOut {
+
+    /// <summary>
+    /// A wrapper around <see cref="StreamWriter"/> used to write Markdown text to a file.
+    /// </summary>
+    public class MdWriter : IDisposable {
+
+        private StreamWriter stream;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MdWriter"/> class, opening a stream to the
+        /// file at the specified path.
+        /// </summary>
+        /// <param name="path">The full path to the output file, including file extension.</param>
+        /// <param name="append">
+        /// If true, output is appended to the file's existing contents; otherwise, the file is
+        /// overwritten.
+        /// </param>
+        public MdWriter(string path, bool append = false) {
+            stream = new StreamWriter(path, append);
+        }
+
+        #region IDisposable
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting
+        /// unmanaged resources.
+        /// </summary>
+        public void Dispose() {
```

Comment:
https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1063#:~:text=CA1063%3A%20Implement%20IDisposable%20correctly%20%20%20%20,is%20breaking%20or%20non-breaking%20%20%20Non-breaking%20

Reply:
I remember this from class... So I want to do: 
`protected virtual void Dispose(bool disposing)
    {
        if (isDisposed) return;
        if (disposing)
        {
            // free managed resources
            stream.Dispose();
        }

        // free native resources if there are any: none in this case
        isDisposed = true;
    }`

And there doesn't need to be a null check on the StreamWriter (like as the code stands) as you can't have a MdWriter with a null stream attribute? MarkdownOut was a Dotnet Framework project I found on Github. I need to comment the name of the original author.

Reply:
I then got rid of the Close Method override, and then put the MdWriter in a using statement in the program. It looks like Close() calls the dispose method in Stream classes and both cause the buffer to be written. Is there any reason to call close if in a using statement?